### ISSUE TITLE
Find a good treadmill to buy and install in the Vicon room 

### ISSUE THREAD

**Issue by DanielePucci** [loc2/axis_bipedal-locomotion#1]
For walking experiments and running experiments (those for the [per](https://github.com/loc2/element_per-at-loc2), for instance), we'll need most probably a treadmill so as repeatability.

We then have to check what the best option is.

@S-Dafarra and @jeljaik @gabrielenava @pi-q, may you take a look at this? Thanks

**Comment by DanielePucci** [comment]
Deadline:  October the 6th

**Comment by gabrielenava** [comment]
I've found this website that may help:

http://www.treadmillreviews.net/best-treadmill/

**Comment by jeljaik** [comment]
In Crossfit ( 😆 I know, I know ) we like very much this type of threadmill called **TrueForm Runner**. The reason why I want to highlight it is because it's not the _usual threadmill_ that you will find in most gyms, but one that tries to create the _nearest simulation of genuine road running_. This implies that it is **manually** operated by the user and I cite:

> the TrueForm Runner from Samsara Fitness features an innovative curved deck that is controlled 100% by the movements and force of the athlete using it

Further information [here](https://www.roguefitness.com/trueform-runner)

**Comment by S-Dafarra** [comment]
I think we first need to define which are the requirements, like maximum speed, length, maximum slope or similar. For example for the *per* we would need an high speed and relatively short treadmill, while with iCub the maximum speed can be consistently lower but with a pretty long extension.

On the [decathlon](https://www.decathlon.it/C-801880-tapis-roulant) there are several of them.

BTW Why such an hurry? :grin:

**Comment by S-Dafarra** [comment]
>This implies that it is manually operated by the user and I cite:

I think it is a good idea for training but, at least with the *per* I think we need it to set-up an initial velocity.  In addition it weights a couple of kilograms, I don't think it will be able to move it :wink:

**Comment by DanielePucci** [comment]
@jeljaik 
> Might be interesting specially for running experiments if I understood correctly the intentions of the PER project.

**element** Jorh, now they are called **element** not projects :smiley:

(and now you think: what the f**k is an element? :sweat_smile:)

Seriously, we have to think of a treadmill not only for the robot, but also for the humans. What I have in mind is a tele-operated system where the human is sensorized and we send whole-body information to the robot (arms, etc): tele-operation + retargeting. 

So the question is: does it exist a treadmill that can move also **laterally** (not only forward motion)?

**Comment by S-Dafarra** [comment]
>So the question is: does it exist a treadmill that can move also laterally (not only forward motion)?

There are some concepts for VR https://www.youtube.com/watch?v=WBEfOcNTaVA
But they are different animals. I don't see a one for all solution here.

**Comment by DanielePucci** [comment]
I like some of the these. Can we find the price of some of these omnidirectional treadmills? What we can do we buy more than one. Probably one straight, and the other omnidirectional

**Comment by DanielePucci** [comment]
Any news on the price? Can we even buy these treadmills/

**Comment by DanielePucci** [comment]
Found this of the [INDIEGOGO](https://www.indiegogo.com/projects/vue-vr-treadmill-omni-directional-vr-treadmill#/) just to have an indication of the price (they do [crowdfunding](https://en.wikipedia.org/wiki/Crowdfunding) so no real indicator). 

I've requested [VIRTUIX](http://www.virtuix.com/) a quote for the [omnidirectional treadmill](http://www.virtuix.com/products/) 

Also, we shall choose a treadmill asap @S-Dafarra @jeljaik

**Comment by DanielePucci** [comment]
ping @pi-q

**Comment by jeljaik** [comment]
Yeah I did the same with Virtuix :) Let's see who gets lucky first.

**Comment by jeljaik** [comment]
@DanielePucci Also asked [**infinadeck**](http://www.infinadeck.com).

**Comment by pi-q** [comment]
I had a discussion with Andreea from ADVR, she gave me a couple of points to check for choosing a treadmill:
- Having an exact control of the velocity is important for repeatability
- Being able to control the treadmill from a computer interface is a great plus

Apart from that, there are treadmills which are split in two, so that for example one side could be made to run faster than the other side. Would it hold any interest in our case?

**Comment by S-Dafarra** [comment]
I don't know about the split configuration, but the computer interface is definitely a plus.

I was looking at the [per](https://www.youtube.com/watch?v=Wa0amJmJ550) video and they claim it can run at 12 mp/h (~19km/h). Thus we would need at least 20km/h for maximum speed. The most expensive commercial treadmills can reach around 24 km/h, but still I don't think we will be able to perform [this](https://www.youtube.com/watch?v=chPanW0QWhA) kind of experiments. 

As I was saying before, if we want to put iCub on it, it would be better to have a long running deck, but there is another problem: the minimum speed is usually 1km/h. I regret to say that we are not there yet :sob:. But actually [other robots](https://www.youtube.com/watch?v=f2Pyu2HxVUQ) don't go that faster :grin:

**Comment by S-Dafarra** [comment]
I have come out with the following shortlist. 

| Name                                                                                                                     | Running deck | Min Speed | Max Speed | Price |
|--------------------------------------------------------------------------------------------------------------------------|:------------:|-----------|-----------|-------|
| [Elite 5000](https://www.decathlon.it/tapis-roulant-elite-5000-id_8380487.html)                                          | 55 x 152 cm  | 1         | 24        | 3.999 |
| [TX-FITNESS TX 10500 AC HRC](https://www.tapisroulantstore.it/tapis-roulant-motorizzato-tx-fitness-tx-10500-ac-hrc.html) | 53 x 145 cm  | 0.8       | 20        | 1099  |
| [HIGH MUSTER T 810](https://www.tapisroulantstore.it/tapis-roulant-high-muster-t-810.html)                               | 53 x 167 cm  | 1         | 24        | 1189  |

They all have motorized slope selection, but I don't know whether there is a PC interface

**Comment by DanielePucci** [comment]
Really nice job guys! @pi-q is definitely right: it would be important to control the speed from a computer. It seems to me that the right one could be [TX-FITNESS TX 10500 AC HRC](https://www.tapisroulantstore.it/tapis-roulant-motorizzato-tx-fitness-tx-10500-ac-hrc.html). 

May we understand if it possible to interface it with a computer @pi-q? Thanks

**Comment by DanielePucci** [comment]
I think we can split this discussion in two issues: omnidirectional and classical treadmills. Let's discuss here on classical tread mills, and keep discussing about the omnidirectional [here](https://github.com/loc2/axis_bipedal-locomotion/issues/3).

Please update the table in the first comment for classical treadmills to ease readability issues of this thread of discussion

**Comment by pi-q** [comment]
There is this company, [Motek](www.motekforcelink.com), which produces pretty interesting treadmills and products. They were initially designed for rehabilitation, but I think they can be quite powerful for us as well. 

For example the [M-Gait](https://www.motekforcelink.com/product/m-gait/) is a modular treadmill system, which comes with an instrumented treadmill and software for system control and programming, which could also be used for gait analysis.  As a plus, if I understood well, the software is compatible with Vicon.

**Comment by DanielePucci** [comment]
Nice @pi-q Can you further investigate about min and max velocity, price, and API? Thanks!

**Comment by pi-q** [comment]
Sure, I sent an e-mail to the company asking for more details.
